packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -59,11 +83,53 @@ const {
     activeOption.value = firstMatch.id;
   },
 });
+
+const loadingMode = computed(() => {
+  if (typeof props.loadingMode === "object") return props.loadingMode.mode;
+  return props.loadingMode;
+});
+
+const loadMoreButtonLabel = computed(() => {
+  if (typeof props.loadingMode === "object" && props.loadingMode.mode === "button") {
+    return props.loadingMode.label;
+  }
+  return t.value("loadMore");
+});
+
+const loadingScrollOffset = computed(() => {
+  if (typeof props.loadingMode === "object" && props.loadingMode.mode === "lazy") {
+    return props.loadingMode.scrollOffset;
+  }
+  return undefined;
+});
+
+const { vScrollEnd, isScrollEnd } = useScrollEnd({
+  enabled: computed(() => loadingMode.value === "lazy"),
+  loading: computed(() => props.loading),
+  offset: loadingScrollOffset,
+});
+
+watchEffect(() => {
+  if (isScrollEnd.value) emit("loadMore");
+});
+
+const isEmpty = computed(() => props.options.length === 0);
 </script>
 
 <template>
-  <div class="onyx-listbox">
-    <ul v-bind="listbox" class="onyx-listbox__options">
+  <div class="onyx-listbox" :aria-busy="props.loading">
+    <div
+      v-if="props.loading && (!loadingMode || isEmpty)"
```

Comment:
what relevance does loadingMode have here? I thought that "loading" always has priority?
```suggestion
      v-if="props.loading"
```